src/core/packet_builder.c

```diff
@@ -197,6 +197,26 @@ QuicPacketBuilderPrepare(
         Connection->Stats.QuicVersion == QUIC_VERSION_2 ?
             QuicKeyTypeToPacketTypeV2(NewPacketKeyType) :
             QuicKeyTypeToPacketTypeV1(NewPacketKeyType);
+
+    if (Connection->Stats.GreaseBitNegotiated == FALSE &&
+        (Connection->PeerTransportParams.Flags & QUIC_TP_FLAG_GREASE_QUIC_BIT) > 0) {
+        //
+        // Endpoints that receive the grease_quic_bit transport parameter from a peer SHOULD set the QUIC Bit to an unpredictable value
+        // unless another extension assigns specific meaning to the value of the bit.
+        //
+        uint8_t RandomValue;
+        (void) CxPlatRandom(sizeof(RandomValue), &RandomValue);
+        Connection->State.FixedBit = (RandomValue % 2);
+        Connection->Stats.GreaseBitNegotiated = TRUE;
+    }
```

Comment:
This logic should go in `QuicConnProcessPeerTransportParameters` which is called when we receive the transport parameters. Also please try to keep comments to 80 character width.

Reply:
I moved the logic into `QuicConnProcessPeerTransportParameters`, also I'm going to fix all comments with 80 character width!